Assets/Scripts/ButtonScript.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using TMPro;
+using UnityEngine;
+using UnityEngine.Events;
+
+
+
+namespace WayOfLife.View
+{
+
+
+    public class ButtonScript : MonoBehaviour
+    {
+
+        private TextMeshPro txt;
+        private UnityEvent onClick;
+
+        // Start is called before the first frame update
+        void Start()
+        {
+            txt = gameObject.GetComponent<TextMeshPro>();
+        }
+
+        public void Render(string text, UnityEvent onClick, bool disabled)
```

Comment:
dont pass UnityEvent from GameManager
you should just create in this class
public event Action\<buttonType\> action 
and listen to the event on gameManager